mesh/meshdb.go

```diff
@@ -440,43 +428,15 @@ func (m *DB) writeBlock(bl *types.Block) error {
 }
 
 func (m *DB) updateLayerWithBlock(blk *types.Block) error {
-	lm := m.getLayerMutex(blk.LayerIndex)
-	defer m.endLayerWorker(blk.LayerIndex)
-
-	lm.m.Lock()
-	defer lm.m.Unlock()
-
-	ids, err := m.layers.Get(blk.LayerIndex.Bytes())
-	var blockIds []types.BlockID
-	if err != nil {
-		// layer doesn't exist, need to insert new layer
-		blockIds = make([]types.BlockID, 0, 1)
-	} else {
-		blockIds, err = types.BytesToBlockIds(ids)
-		if err != nil {
-			return errors.New("could not get all blocks from database")
-		}
-	}
-	m.Debug("added block %v to layer %v", blk.ID(), blk.LayerIndex)
-
-	blockIds = append(blockIds, blk.ID())
-	types.SortBlockIDs(blockIds)
-	w, err := types.BlockIdsToBytes(blockIds)
-	if err != nil {
-		return errors.New("could not encode layer blk ids")
-	}
-
-	m.layers.Put(blk.LayerIndex.Bytes(), w)
-	return nil
-}
-
-func (m *DB) getLayerHashKey(layerID types.LayerID) []byte {
-	return []byte(fmt.Sprintf("layerHash_%v", layerID.Bytes()))
+	var b bytes.Buffer
+	b.Write(blk.LayerIndex.Bytes())
+	b.Write(blk.ID().Bytes())
+	return m.layers.Put(b.Bytes(), nil)
```

Comment:
this code doesn't seem to preserve previous blocks saved for this layer.

Reply:
what do you mean? i think all previous writes are untouched, e.g. if we do multiple writes such as:
1. layer_01_block_01
2. layer_01_block_02
all of them will be in the store

Reply:
i misread. yes. all others are preserved.